### ISSUE TITLE
How to solve the checkboard artifact in medical image?

### ISSUE THREAD

**Issue by John1231983** [junyanz/pytorch-CycleGAN-and-pix2pix#770]
Currently, I am using cyclegan for image super-resolution. I am using two strategies to solve the checkboard artifact that mentioned in some thread of this project
1. Train longer --> The checkboard still maintains in the image
2. Using up-sampling with bilinear/bicubic interpolation--> The checkboard is gone. But the output image is blurring
3. Shuffle pixel (did not try because it does not support for 5D input likes BxCxDxHxW)

This is an example of blurring using bicubic interpolation (c) and linear interpolation. I expect the method that can obtain the result likes figure e or f. Could you suggest any method for solving the problem? Thanks
![Capture](https://user-images.githubusercontent.com/24875971/65005079-ada96a80-d8cc-11e9-8120-d4b421a8992d.JPG)

**Comment by junyanz** [comment]
The generator and the discriminator used in super-resolution might be different from cyclegan networks. See some prior [work](http://openaccess.thecvf.com/content_cvpr_2018_workshops/papers/w13/Yuan_Unsupervised_Image_Super-Resolution_CVPR_2018_paper.pdf) on using cyclegan for super-resolution tasks. I will start from that and then try to improve it. You may also borrow the network architecture from other recent super-resolution papers.

**Comment by John1231983** [comment]
Thanks. I solved it by using NN upsampling. The deconv does not help when we train longer.  When I train the network with more epoch and learning rate, I found that the checkboard artifact will appear more. Does it mean the model overfitting or collapse? How to know the model collapse? Thanks

**Comment by junyanz** [comment]
I am glad that you figured it out. You can evaluate your model on a hold-out validation set, and see if it still works.

**Comment by John1231983** [comment]
@junyanz : it still worked but my problem is that when I train the model with more epochs, the performance reduces. Should I use the early stopping?  Note that, all loss do not increase if I train with more epoch, but the accuracy is reduced

**Comment by junyanz** [comment]
You should do early stopping or reduce the learning rate.

**Comment by jiangxuetao0823** [comment]
> Thanks. I solved it by using NN upsampling. The deconv does not help when we train longer. When I train the network with more epoch and learning rate, I found that the checkboard artifact will appear more. Does it mean the model overfitting or collapse? How to know the model collapse? Thanks

Do you use the nearest neighbor up-sampling + convolution to replace deconv ?

**Comment by John1231983** [comment]
Yes. But the result becomes blurring. It may be trade-off. Let me know if you also have same issue

**Comment by jiangxuetao0823** [comment]
> Yes. But the result becomes blurring. It may be trade-off. Let me know if you also have same issue

Yes, I also used the nearest neighbor up-sampling + convolution to replace up-sampling. Although it can overcome the checkerboard problem, but the overall brightness of the image is not uniform, coutour will appear.
